src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVKmerizer.java

```diff
@@ -12,23 +12,23 @@
  * Iterator over successive Kmers from a sequence of characters.
  * Silently skips over parts of the sequence that has characters other than A, C, G, or T.
  */
-public class SVKmerizer implements Iterator<SVKmer> {
+public class SVKmerizer<KType extends SVKmerBase> implements Iterator<KType> {
```

Comment:
I think my muddled thinking sent Mark in the wrong direction.  I agree that this class doesn't need to be generic -- it can just deal with SVKmerBase.  Serialization should work fine -- it finds the correct serializer based on the actual class.